task-list/app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,53 @@
+class TasksController < ApplicationController
+
+
+  def index   #this is enough for the controller to render a view, but need to tell rails when this is an appropriate action
+    @all_tasks = all_incomplete_tasks
+  end
+
+  def all_incomplete_tasks
+    Task.where(completed_at: nil)
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    render :single_task
+  end
```

Comment:
This is a good use of explicitly saying "render :some_view", since it has a different name than the function